packages/react-components/lib/tasks/create-task.tsx

```diff
@@ -296,7 +335,7 @@ function DeliveryTaskForm({
           options={[]}
           onChange={(_ev, newValue) =>
```

Comment:
[chrome-capture-2023-8-7.webm](https://github.com/open-rmf/rmf-web/assets/37310205/b8693589-5465-437f-af7b-51bd59d1c8cc)

I found this. I don't know if it was intentional. But by completing all the fields but then deleting both pickup location or dropoff location the button is enabled because it retains the previous value. Maybe that is not visually correct. This happens in the DeliveryTaskForm function.
I leave something that may help but definitely can be improved. 

   ```tsx
       onChange={(_ev, newValue) => {
            if (newValue !== null && dropoffPoints[newValue]) {
              onInputChange({
                ...taskDesc,
                dropoff: {
                  ...taskDesc.dropoff,
                  place: newValue,
                  handler: dropoffPoints[newValue],
                },
              });
            } else {
              onInputChange({
                ...taskDesc,
                dropoff: {
                  ...taskDesc.dropoff,
                  place: '', 
                  handler: '', 
                },
              });
            }
          }
```
The example is for dropoff-location
It is likely that the same should be done for pickup-location

Reply:
good catch! I've fixed it for the other task forms and did some more refactoring